src/middlewared/middlewared/test/integration/assets/roles.py

```diff
@@ -0,0 +1,26 @@
+import errno
+import pytest
+
+from middlewared.client.client import ClientException
+from middlewared.test.integration.assets.account import unprivileged_user_client
+
+
+def common_checks(
+    method, role, valid_role, valid_role_exception=True, method_args=None, method_kwargs=None, is_return_type_none=False
+):
+    method_args = method_args or []
+    method_kwargs = method_kwargs or {}
+    with unprivileged_user_client(roles=[role]) as client:
+        if valid_role:
```

Comment:
On the valid_role branch should we confirm `errno` is NOT `EACCES` and `error` is NOT 'Not authorized'?

Reply:
@mgrimesix we can just check that the exception is not `ClientException` and that should be sufficient as other types of errors are raised when there is an issue with malformed parameters etc